src/models/service/balances.rs

```diff
@@ -0,0 +1,26 @@
+use crate::providers::info::{TokenInfo, DefaultInfoProvider};
+use serde::Serialize;
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Balance {
+    pub token_info: TokenInfo,
+    pub amount: String,
+    pub fiat_amount: String,
+}
+
+impl Fiat {
+    pub fn from_dasd(fiat: &str, info_provider: &DefaultInfoProvider) -> Self {
+        match fiat {
+            "usd" => Fiat::Usd(1.0),
+            "eur" => Fiat::Eur(info_provider.exchange_usd_to_eur().unwrap_or(1.0)),
```

Comment:
Yeah, I was not sure about the spec, I thought it was just Euro and Usd. That's why I decided to wrap stuff in our own enum in case someday we decided to change exchange rate values provider. I will just pass on the currency code then, it should simplify things.